Makefile

```
@@ -29,42 +29,41 @@ LDFLAGS=-ldflags "-w -s -X main.Version=${TAG}"
 .PHONY: install
 install:
 	@echo ....... Creating namespace ....... 
-	- oc new-project ${NAMESPACE}
+	- kubectl create namespace ${NAMESPACE}
 	@echo ....... Applying Mobile Security Service CRDS and Operator .......
-	- kubectl apply -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_crd.yaml
-	- kubectl apply -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservicedb_crd.yaml
-	- kubectl apply -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityserviceapp_crd.yaml
-	- kubectl apply -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservicebackup_crd.yaml
+	- kubectl apply -n ${NAMESPACE} -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_crd.yaml
+	- kubectl apply -n ${NAMESPACE} -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservicedb_crd.yaml
+	- kubectl apply -n ${NAMESPACE} -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityserviceapp_crd.yaml
+	- kubectl apply -n ${NAMESPACE} -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservicebackup_crd.yaml
 	@echo ....... Applying Rules and Service Account .......
-	- kubectl apply -f deploy/cluster_role.yaml
-	- kubectl apply -f deploy/cluster_role_binding.yaml
-	- kubectl apply -f deploy/service_account.yaml
+	- kubectl apply -n ${NAMESPACE} -f deploy/cluster_role.yaml
+	- kubectl apply -n ${NAMESPACE} -f deploy/cluster_role_binding.yaml
+	- kubectl apply -n ${NAMESPACE} -f deploy/service_account.yaml
 	@echo ....... Applying Mobile Security Service Operator .......
-	- kubectl apply -f deploy/operator.yaml
+	- kubectl apply -n ${NAMESPACE} -f deploy/operator.yaml
 	@echo ....... Creating the Mobile Security Service and Database .......
-	- kubectl apply -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_cr.yaml
-	- kubectl apply -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservicedb_cr.yaml
+	- kubectl apply -n ${NAMESPACE} -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_cr.yaml
+	- kubectl apply -n ${NAMESPACE} -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservicedb_cr.yaml
 	@echo ....... Creating namespace ${APP_NAMESPACES} .......
-	- oc new-project ${APP_NAMESPACES}
+	- kubectl create namespace ${APP_NAMESPACES}
 
 .PHONY: uninstall
 uninstall:
 	@echo ....... Uninstalling .......
-	- oc project ${NAMESPACE}
 	@echo ....... Deleting the Mobile Security Service and Database .......
-	- kubectl delete -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_cr.yaml
-	- kubectl delete -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservicedb_cr.yaml
+	- kubectl delete -n ${NAMESPACE} -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_cr.yaml
```

```suggestion
	- kubectl delete MobileSecurityServiceApp -n ${NAMESPACE} --all
	- kubectl delete MobileSecurityServiceApp -n ${APP_NAMESPACES} --all
	- kubectl delete -n ${NAMESPACE} -f deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservice_cr.yaml
```

I think this should avoid the issue you're facing. By deleting any app CRs first, the operator should be able to perform the cleanup logic in the service. However, if you delete the service first, then the operator can't complete the logic it needs before removing the finalizer (and you need to manually remove them, like you're highlighting in the README), and when you get to trying to delete the app CRD it hangs because there are still CRs.

---

It is a specific command very useful to check and test how to delete the app example CR. The install/uninstall is to do it for the service/operator ( database and MSS). Regards the app CR it is an example which following the README shows how to apply and delete to check its working as the next step. However, it is optional since we can have 0 to N APP CR applied in N namespaces.